src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -0,0 +1,137 @@
+   * @return True if any of the available strings can be resolved with the passed string.
+   */
+  def isResolved(
+      spark: SparkSession,
+      requiredString: String,
+      availableStrings: Iterable[String]): Boolean = {
+    availableStrings.exists(isResolved(spark, requiredString, _))
+  }
+
+  /**
+   * Resolves ALL of the requiredStrings with availableStrings based on spark's resolver. Returns
+   * true if all the requiredStrings can be resolved with some string from availableStrings.
+   * Returns false otherwise.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param availableStrings List of availble strings to resolve from.
+   * @return True if every one of requiredStrings can be resolved from availableStrings. False
+   *         otherwise.
+   */
+  def isResolved(
+      spark: SparkSession,
+      requiredStrings: Iterable[String],
+      availableStrings: Iterable[String]): Boolean = {
+    requiredStrings.forall(isResolved(spark, _, availableStrings))
+  }
+
+  /**
+   * Return available string if required string can be resolved with it, based on spark resolver.
+   *
+   * @param spark Spark Session.
+   * @param requiredString The string that requires resolution.
+   * @param availableString Available list of strings to resolve from.
+   * @return Optional available string if resolution is successful, else None
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredString: String,
+      availableString: String): Option[String] = {
+    if (isResolved(spark, requiredString, availableString)) Some(availableString) else None
+  }
+
+  /**
+   * Finds the first matching resolved string from the list of availableStrings, when resolving
+   * for requiredString. If no matching string found, return None.
+   *
+   * @param spark Spark session.
+   * @param requiredString The string that requires resolution.
+   * @param availableStrings All available strings to find a match from.
+   * @return First matching (i.e. resolved) string from availableStrings. If no match is found,
+   *         return None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredString: String,
+      availableStrings: Iterable[String]): Option[String] = {
+    availableStrings.find(isResolved(spark, requiredString, _))
+  }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the list of availableStrings. Returns a
+   * sequence of Optional values for matches, None for unmatched strings.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param availableStrings List of available strings to resolve from.
+   * @return Sequence of optional values of resolved strings or None's.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Iterable[String],
+      availableStrings: Iterable[String]): Seq[Option[String]] = {
+    requiredStrings.map(resolve(spark, _, availableStrings)).toSeq
```

Comment:
This API is a bit confusing since the behavior is different from
```
  def isResolved(
      spark: SparkSession,
      requiredStrings: Iterable[String],
      availableStrings: Iterable[String]): Boolean = {
    requiredStrings.forall(isResolved(spark, _, availableStrings))
  }
```
Can we just return `Option[Seq[String]]`? Do we have a case where we want to match partially?

Reply:
done, thanks @imback82